src/SolverInterface/DspOsi.h

```diff
@@ -34,9 +40,47 @@ class DspOsi {
 		throw CoinError("Quadratic objective is not supported.", "loadQuadraticObjective", "DspOsi");
 	}
 
+	/** load quadratic constrs */
+	virtual void addQuadraticRows(int nqrows, int * linnzcnt, int * quadnzcnt, double * rhs, int * sense, int const ** linind, double const ** linval, 
+										int const ** quadrow, int const ** quadcol, double const ** quadval) {};
+
+	/** throw error */
+	virtual inline void checkDspOsiError(int err, std::string solverfuncname, std::string dsposimethod){};
+
+	/** write problem file */
+	virtual void writeProb(char const * filename_str, char const * filetype_str) {};
+
+	/** set problem type */
+	virtual void setProbType(bool isqp, bool isqcp) {
+		
+		if (si_->getNumIntegers() > 0)
+			ismip_ = true;
+		
+		isqp_ = isqp;
+		isqcp_ = isqcp;
+	}
+
+	/** change problem type to MIQCP */
+	virtual void switchToMIQCP(void){};
+
+	/** change problem type to MIQP */
+	virtual void switchToMIQP(void){};
+
+	/** change problem type to QCP */
+	virtual void switchToQCP(void){};
+
+	/** change problem type to QP */
+	virtual void switchToQP(void){};
+
 	/** solve problem */
 	virtual void solve() = 0;
 
+	/** modify OsiSolverInterface::initialSolve */
+	virtual void QCQPSolve(){};
+
+	/** modify OsiSolverInterface::branchAndBound */
+	virtual void MIQCQPSolve(){};
+
```

Comment:
I think these may better be specific to solver.

Reply:
Updated accordingly.